### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Chaseknis**
# In this PR we accomplished the following:
- Remove current_user method in ApplicationController
- Install and setup devise so that the user can do the following:

> _Can register a new user._
> _User logs in with a combination of email and password._
> _Hashed password should be stored in the database._
> _Ask for confirmation of email._
> _Can reset password._

- Update posts_controller.rb and users_controller.rb Update post.rb and user.rb models
- Update views directory
- Update schema.rb
- Add 20230519144300_add_devise_to_users.rb migration file Run linters for errors

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @Chaseknis,

terminator

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair-programming project but there aren't commits from your partner. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Chaseknis** [comment]
@franclobo We are sorry, we got carried away. We have solved that issue now
.

File: app/models/user.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (ridwanediallo) [inline app/models/user.rb]: - [ ] Good job adding devise to your project. However, as stated in the requirements, please ensure to add the feature to **ask for email confirmation**.
To accomplish this task, you need to enable **:confirmable**  in the user model. Like below :point_down: 
```
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable, :confirmable
```

#### Requirement :point_down: 
Screenshot from 2023-05-19 19-11-15

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
```
Comment (ridwanediallo) [inline app/views/devise/registrations/new.html.erb]: - [ ] Please ensure that the form for registering a user matches the one provided in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_login.png), as stated in the requirement.

#### Requirement :point_down: 
Screenshot from 2023-05-19 19-28-14

### Your form so far 
Screenshot from 2023-05-19 19-18-43

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
```
Comment (ridwanediallo) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please ensure that the form for login a user matches the one provided in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_login.png), as stated in the requirement.

#### Requirement :point_down: 
Screenshot from 2023-05-19 19-28-14

### Your form so far 
Screenshot from 2023-05-19 19-18-25

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi team, 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your PR looks professional. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @Chaseknis 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested change implemented.✅ 


## Optional suggestions

- I think it will be better if you include a gem that will help users confirm there email locally for testing purposes e.g `letter-opener` gem

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chaseknis** [comment]
thank you @vickymarz

**Pull request merged**